packages/maskbook/src/plugins/ITO/UI/ITO.tsx

```diff
@@ -150,6 +150,30 @@ const TokenItem = ({ price, token, exchangeToken }: TokenItemProps) => {
     )
 }
 
+function _format(amount: BigNumber, token: EtherTokenDetailed | ERC20TokenDetailed): string {
+    if (amount.isZero()) {
+        return '0'
+    }
+    const _amount = formatBalance(amount, token.decimals)
+    const [number, decimal] = _amount.split('.')
+
+    if (!decimal) return number
+
+    if (number.length <= 6) {
+        if (decimal.length >= 6) {
+            return `${number}.${decimal.slice(0, 6)}`
+        }
+        return `${number}.${decimal.padEnd(6, '0')}`
```

Comment:
```ts
const balance = formatBalance(amount, token.decimals)
const sd = balance.precision(true)
return balance.toFixed(sd.length <= 12 ? clamp(sd.length, 6, 12) : 0)
```

see https://mdn.io/Math/toFixed
see https://lodash.com/docs/4.17.15#clamp
see https://mikemcl.github.io/bignumber.js/#toFix

Reply:
the sample need test and verify.

Reply:
```
> test = (input) => { const balance = new BigNumber(input); return (balance.precision(true) - balance.decimalPlaces()) <= 11 ? balance.toPrecision(11) : balance.toFixed(0) }
[Function: test]
> test('123456.78999')
'123456.78999'
> test('123456.789')
'123456.78900'
> test('123456.7')
'123456.70000'
> test('123456000.7')
'123456000.70'
> test('123456000000.7')
'123456000001'
```

Reply:
ok

Reply:
balance.precision() ： Number of significant digits